### PR TITLE
fix memory leaks in recentActions and CameraUploads

### PR DESCRIPTION
**Pull request by simonwng**
In iOS 5.4 build, we have detected big memory leaks in recent actions and some obvious small leaks in camera uploads. This PR is to fix the leaks.

### CODE REVIEW

**Review by jnavarrom — APPROVED** [review]

File: bindings/ios/MEGASdk.mm [hunk 1]
```diff
@@ -2145,6 +2146,8 @@ - (NSMutableArray *)recentActions {
         [recentActionBucketMutableArray addObject:recentActionBucket];
     }
     
+    delete megaRecentActionBucketList;
```
Comment (joel-meng-nz) [inline bindings/ios/MEGASdk.mm:2149]: wow, if have to manually deallocate the objects like this, there are more, at least the banner API binding, need to be updated as well.

Comment (jnavarrom) [reply bindings/ios/MEGASdk.mm:2149]: It will depend, you need to do it if the SDK saids: "* You take the ownership of the returned value."

Comment (sergiohs84) [reply bindings/ios/MEGASdk.mm:2149]: If the SDK documentation states that you take the ownership of the returned object, @joel-meng-nz , you do need to deallocate them, yes.

Is the SmartBanner feature included in 5.4? If not, please, fix your bindings in a different branch/PR. Thanks

Comment (sergiohs84) [reply bindings/ios/MEGASdk.mm:2149]: Navarro already told me it does not. Merging this PR...

**Pull request merged**